### DISCUSSION TITLE
Format Strings: Return no module if parsing the version failed?

### DISCUSSION THREAD

**Discussion by ghost** [starship/starship#1435] [Ideas]
This is related to https://github.com/starship/starship/pull/1374#pullrequestreview-442702580.
There is said:
> We still have some modules where the call to the specific program is not computed inside of the format statement. Which means that if someone disables the version variable the version is still computed.
> I will open a PR to solve that.

Now I came to a question that I cannot answer myself:
Should we

1. return no module if we fail to parse the version (eg. the program from which we need the version is not installed) or
2. still show the symbol so that the user knows that he is in a project of that type but does not have the correct utilities installed.

I'd like to have some input from everyone since both option seem pretty good to me.

**Comment by matchai** [comment]
Good question!
I would be in favor of the second option.
Much like how we handle flags, I think we should endeavor to give as much context as possible given what is available to us at the time.
